bloom-housing/bloom
# feat: replace UIC `GridSection` with Seeds `Grid`

## Issue Overview

This PR addresses #3575 


## Description

Apologies in advance for such a large changeset! 😅

This PR provides the original intention of replacing all usage of UIC `GridSection` with Seeds `Grid`. In addition it uses Seeds `Card` for some drawer content instead of bespoke Tailwind classes, and `Heading/HeadingGroup` from Seeds. It also adds a few local utility classes based directly on Seeds tokens (and we can investigate upstreaming those to Seeds as a separate PR).

* Note: to make it easier to swap out `GridSection`, I created a local component called `SectionWithGrid` which then directly consumes Seeds grid and heading components. If you have any suggestions for a better name or otherwise how that works now, feel free to comment in review!

## How Can This Be Tested/Reviewed?

Visit the preview site: https://deploy-preview-3643--partners-bloom-dev.netlify.app/
and view/edit some listings, and see how they compare visually with the same screens on the live Partners dev site.

There are a few typographic updates (primarily headings) to align with Seeds typography. I also slightly changed the layout of the top of the Listing detail page grid for better readability. Otherwise, if there are different configurations of rows/columns in grid sections or missing fields or whatever, we should assume that may be a bug and investigate.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
